Sources/BuildServerIntegration/BuildServerManager.swift

```diff
@@ -962,6 +962,96 @@ package actor BuildServerManager: QueueBasedMessageHandler {
     return locations.map { locationAdjustedForCopiedFiles($0) }
   }
 
+  package func workspaceEditAdjustedForCopiedFiles(_ workspaceEdit: WorkspaceEdit?) async -> WorkspaceEdit? {
+    guard var edit = workspaceEdit else {
+      return nil
+    }
+    if let changes = edit.changes {
+      var newChanges: [DocumentURI: [TextEdit]] = [:]
+      for (uri, edits) in changes {
+        let newUri = self.locationAdjustedForCopiedFiles(
+          Location(uri: uri, range: Position(line: 0, utf16index: 0)..<Position(line: 0, utf16index: 0))
```

Comment:
Instead of synthesizing a `Location` with an irrelevant position, I think we should just have a function that adjusts a `DocumentURI` for copied files.